docs/rfcs/xxxx-river.md

```diff
@@ -0,0 +1,399 @@
+# River: A Flow-optimized config language
+
+* Date: 2022-06-27
+* Author: Robert Fratto (@rfratto), Matt Durham (@mattdurham)
+* PR: [grafana/agent#1839](https://github.com/grafana/agent/pull/1839)
+* Status: Draft
+
+## Summary
+
+Grafana Agent developers have been working towards a feature called Grafana
+Agent Flow ([RFC-0004][]), a component-based re-imagining of Grafana Agent
+which compartmentalize the different configurable pieces of the agent, allowing
+users to more easily understand and debug configuration issues. Grafana Agent
+Flow was purposefully scoped broadly to allow for exploring many different
+component-based approaches for prototyping the experimental feature.
+
+The current implementation strategy focuses around expressions: settings for
+components can be derived from expressions which can reference and mutate the
+outputs of other components. Values can refer to arbitrary Go values like
+interfaces or channels, enabling component developers to easily allow users to
+construct data pipelines using Go APIs without requiring knowledge of the
+underlying implementation.
+
+The initial expressions prototype used [HCL][], which initially fit Flow's
+needs during early prototyping. However, the growing dependency on passing
+around arbitrary Go values to build pipelines started to conflict with the
+limitations of HCL, making HCL increasingly insufficient for Flow's specific
+use case.
+
+We examined alternatives to HCL such as YAML, CUE, Jsonnet, Lua, and Go itself.
+Eventually, we determined that the way we use arbitrary Go values in
+expressions for constructing pipelines is a new use case warranting a
+custom-built language.
+
+This document proposes River, an HCL-inspired declarative expressions-based
+language for continuous runtime evaluation. The decision to propose a new
+language is not taken lightly, and is seen as the last resort. As such, much of
+this proposal will focus on the rationale leading to this choice.
+
+## Goals
+
+* Minimize learning curve as much as possible to reduce friction
+* Make it easy for developers to consume for Flow components
```

Comment:
This sentence doesn't flow. Is it 'Make it easy for developers to write flow components'? Or for users to easily use flow components?

Reply:
oops, not sure what happened here. fixed